Tools improvements

d2s opened the pull request:
## What is the goal of this PR?

- Find out potential code reliability issues faster
- Improve consistency of future improvements
- Improve documentation

## How does this PR achieve the goal?

### chore: add EditorConfig settings

- Add EditorConfig settings

#### Reason for adding

"EditorConfig helps maintain consistent coding styles
for multiple developers working on the same project
across various editors and IDEs."

#### More details

- <https://editorconfig.org>



### docs: improve Markdown structure

- Improve Markdown structure
- Solve `markdownlint` warnings
  - `MD040/fenced-code-language`: Fenced code blocks should have a language specified
- Add inline code highlights



### chore: improve .gitignore patterns

- Improve `.gitignore` patterns



### chore: add ESLint & Prettier tooling

- Add ESLint & Prettier tooling
- Add ESLint configuration that is compatible with Prettier
- Add npm run tasks for running the tools
  - `npm run eslint`
    - Checks files under `bin/` directory for structure issues
  - `npm run prettier`
    - Autoformats code style for the files
- Update `package-lock.json` lockfile

#### Reason for adding

- Find out potential code reliability issues faster
- Improve consistency of future improvements

#### Command used for adding the dependencies

```sh
npm install --save-dev --save-exact eslint eslint-config-prettier eslint-plugin-prettier
npm install --save-dev --save-exact prettier
```

#### More details

- <https://prettier.io>
- <https://eslint.org>

d2s committed:
chore: add EditorConfig settings

- Add EditorConfig settings

Reason for adding:

"EditorConfig helps maintain consistent coding styles
for multiple developers working on the same project
across various editors and IDEs."

More details:

- https://editorconfig.org

d2s committed:
docs: improve Markdown structure

- Improve Markdown structure
- Solve `markdownlint` warnings
  - `MD040/fenced-code-language`: Fenced code blocks should have a language specified
- Add inline code highlights

d2s committed:
chore: improve .gitignore patterns

- Improve `.gitignore` patterns

d2s committed:
chore: add ESLint & Prettier tooling

- Add ESLint & Prettier tooling
- Add ESLint configuration that is compatible with Prettier
- Add npm run tasks for running the tools
  - `npm run eslint`
    - Checks files under `bin/` directory for structure issues
  - `npm run prettier`
    - Autoformats code style for the files
- Update `package-lock.json` lockfile

Reason for adding:

- Find out potential code reliability issues faster
- Improve consistency of future improvements

Command used for adding the dependencies:

npm install --save-dev --save-exact eslint eslint-config-prettier eslint-plugin-prettier
npm install --save-dev --save-exact prettier

More details:

- https://prettier.io
- https://eslint.org

Mulperi commented on .eslintrc.js:
```diff
@@ -0,0 +1,20 @@
+  },
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: "module",
```
Should this actually be "script"?

Mulperi commented on .gitignore:
```diff
@@ -1 +1,26 @@
+# Local package management cache
 node_modules/
+
+# Ignore Yarn lockfile because we you npm
```
we you?

Mulperi commented on .eslintrc.js:
```diff
@@ -0,0 +1,20 @@
+    node: true,
+  },
+  globals: {
+    Atomics: "readonly",
+    SharedArrayBuffer: "readonly",
```
Do we need these?

d2s replied:
Most likely not. Removed those lines...

d2s committed:
fix: improve ESLint config

- Improve ESLint config
- Reduce amount of lines
- Set `sourceType` to `script`

d2s committed:
fix: improve spelling

- Improve spelling

d2s commented:
Updated the pull request with the requested changes.
